llvm/lib/SandboxIR/SandboxIR.cpp

```diff
@@ -262,6 +262,113 @@ const char *Instruction::getOpcodeName(Opcode Opc) {
   llvm_unreachable("Unknown Opcode");
 }
 
+llvm::Instruction *Instruction::getTopmostLLVMInstruction() const {
+  Instruction *Prev = getPrevNode();
+  if (Prev == nullptr) {
+    // If at top of the BB, return the first BB instruction.
+    return &*cast<llvm::BasicBlock>(getParent()->Val)->begin();
+  }
+  // Else get the Previous sandbox IR instruction's bottom IR instruction and
+  // return its successor.
+  llvm::Instruction *PrevBotI = cast<llvm::Instruction>(Prev->Val);
+  return PrevBotI->getNextNode();
+}
+
+BBIterator Instruction::getIterator() const {
+  auto *I = cast<llvm::Instruction>(Val);
+  return BasicBlock::iterator(I->getParent(), I->getIterator(), &Ctx);
+}
+
+Instruction *Instruction::getNextNode() const {
```

Comment:
is this necessary? this is part of `ilist_node_with_parent`, so the name doesn't really make sense here. if we do want to support this operation, I'd at least rename it to something like `getNextInstruction()`

Reply:
Well `getNextNode()` is part of the Instruction API, so the user expects it to be there. So I am not sure we should remove it, or use a different name for a function that does the same thing. Perhaps we could have both `getNextInstruction()` and `getNextNode()`? Wdyt?

Reply:
I still think that `Node` is an implementation detail and we can just have `getNextInstruction()`, but don't feel that strongly. This is fine for now.